apps/server/src/modules/files-storage/entity/filerecord.entity.spec.ts

```diff
@@ -158,4 +158,262 @@ describe('FileRecord Entity', () => {
+
+				expect(result).toBe(true);
+			});
+		});
+
+		describe('WHEN file record security status is not BLOCKED', () => {
+			const setup = () => {
+				const fileRecord = fileRecordFactory.build();
+
+				fileRecord.securityCheck.status = ScanStatus.VERIFIED;
+
+				return { fileRecord };
+			};
+
+			it('should return false', () => {
+				const { fileRecord } = setup();
+
+				const result = fileRecord.isBlocked();
+
+				expect(result).toBe(false);
+			});
+		});
+	});
+
+	describe('isPending is called', () => {
+		describe('WHEN file record security status is PENDING', () => {
+			const setup = () => {
+				const fileRecord = fileRecordFactory.build();
+
+				fileRecord.securityCheck.status = ScanStatus.PENDING;
+
+				return { fileRecord };
+			};
+
+			it('should return true', () => {
+				const { fileRecord } = setup();
+
+				const result = fileRecord.isPending();
+
+				expect(result).toBe(true);
+			});
+		});
+
+		describe('WHEN file record security status is not PENDING', () => {
+			const setup = () => {
+				const fileRecord = fileRecordFactory.build();
+
+				fileRecord.securityCheck.status = ScanStatus.VERIFIED;
+
+				return { fileRecord };
+			};
+
+			it('should return false', () => {
+				const { fileRecord } = setup();
+
+				const result = fileRecord.isPending();
+
+				expect(result).toBe(false);
+			});
+		});
+	});
+
+	describe('isVerified is called', () => {
+		describe('WHEN file record security status is VERIFIED', () => {
+			const setup = () => {
+				const fileRecord = fileRecordFactory.build();
+
+				fileRecord.securityCheck.status = ScanStatus.VERIFIED;
+
+				return { fileRecord };
+			};
+
+			it('should return true', () => {
+				const { fileRecord } = setup();
+
+				const result = fileRecord.isVerified();
+
+				expect(result).toBe(true);
+			});
+		});
+
+		describe('WHEN file record security status is not VERIFIED', () => {
+			const setup = () => {
+				const fileRecord = fileRecordFactory.build();
+
+				fileRecord.securityCheck.status = ScanStatus.BLOCKED;
+
+				return { fileRecord };
+			};
+
+			it('should return false', () => {
+				const { fileRecord } = setup();
+
+				const result = fileRecord.isVerified();
+
+				expect(result).toBe(false);
+			});
+		});
+	});
+
+	describe('setSecurityCheck is called', () => {
+		describe('WHEN file record with already created default pending status exists', () => {
+			const setup = () => {
+				const fileRecord = fileRecordFactory.build({});
+				fileRecord.securityCheck.status = ScanStatus.PENDING;
+
+				return { fileRecord };
+			};
+
+			it('should be update the securityCheck in fileRecord by set a new on.', () => {
+				const { fileRecord } = setup();
+				const securityCheck = new FileSecurityCheck({ status: ScanStatus.BLOCKED });
+
+				fileRecord.setSecurityCheck(securityCheck);
+
+				expect(fileRecord.securityCheck).toEqual(securityCheck);
+			});
+
+			it('should not be do nothing if passing element is not valid instance', () => {
```

Comment:
Das sind viele Verneinungen ^^

Ich denke, es sollte heißen: 'should do nothing if argument is not an instance of FileSecurityCheck'.

Reply:
ändere ich noch